### PR TITLE
IOSAS-856 | Defect | The early logout issue fix

### PR DESCRIPTION
**Pull request by fwpushan**
- Fix the early logout issue. 
- Add more logs

### CODE REVIEW

File: backend/src/components/auth.js [hunk 1]
```diff
@@ -8,26 +8,39 @@ const qs = require('querystring');
...
+  isTokenExpired(token, context = 'NA') {
     const now = Date.now().valueOf() / 1000;
     const payload = jsonwebtoken.decode(token);
+    log.info('isTokenExpired: context: ', context);
+    if (payload['exp']) {
+      // log.info('Token: JWT', token)
+      // log.info('isTokenExpired: exp:', payload['exp']);
+      // log.info('isTokenExpired: now: ', now + 30);
+      // log.info('isTokenExpired: diff: ', payload['exp'] - (now + 30));
+      const expDate = new Date((payload['exp'] * 1000));
```
Comment (mmumby) [inline backend/src/components/auth.js:21]: Can you remove all commented out logs

File: backend/src/components/auth.js [hunk 2]
```diff
@@ -8,26 +8,39 @@ const qs = require('querystring');
...
   },
 
   // Check if JWT Refresh Token has expired
-  isRenewable(token) {
+  isRenewable(token, context = 'NA') {
     const now = Date.now().valueOf() / 1000;
     const payload = jsonwebtoken.decode(token);
-
+    if (!payload.iat || typeof payload.iat === 'undefined') {
+      log.warn('isRenewable', '| The token has no IAT possibly wrong token', payload);
```
Comment (mmumby) [inline backend/src/components/auth.js:33]: you should be able to just call `payload.iat === undefined`, unless its returning a string of undefined

File: backend/src/components/auth.js [hunk 3]
```diff
@@ -36,24 +49,31 @@ const auth = {
...
       );
 
       log.verbose('renew', utils.prettyStringify(response.data));
-      if (response && response.data && response.data.access_token && response.data.refresh_token) {
+      if (
+        response &&
+        response.data &&
+        response.data.access_token &&
+        response.data.refresh_token
+      ) {
```
Comment (mmumby) [inline backend/src/components/auth.js:72]: you can simpliify with the ? mark null check

' response?.data?.access_token &&
        response?.data?.refresh_token'

**Review by mmumby — APPROVED** [review]
Looks good! Glad to have this fixed. Can you please remove all the commented out logs in every file before merging!

File: backend/src/components/auth.js (hunk 2 above)
Comment (fwpushan) [reply backend/src/components/auth.js:33]: Try to address the null value. Here value could be 0.

Comment (mmumby) [reply backend/src/components/auth.js:33]: 0 is valid in this situation? 

!payload.iat will catch 0, null, undefined, and false. 

Is this what you're trying to accomplish?
If (!payload.iat && payload.iat !== 0)

Comment (fwpushan) [reply backend/src/components/auth.js:33]: 0 is valid

**Pull request merged**